homeassistant/components/advantage_air/climate.py

```diff
@@ -188,6 +210,54 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
                 }
             )
 
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode."""
+        change = {}
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            change[ADVANTAGE_AIR_MYTEMP_ENABLED] = preset_mode == ADVANTAGE_AIR_MYTEMP
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            change[ADVANTAGE_AIR_MYAUTO_ENABLED] = preset_mode == ADVANTAGE_AIR_MYAUTO
+        await self.async_update_ac(change)
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to state changes if required."""
+        await super().async_added_to_hass()
+        # If presets are supported
+        if self._attr_supported_features & ClimateEntityFeature.PRESET_MODE:
+            # Track state changes, as preset changes require a reload
+            self.async_on_remove(
+                async_track_state_change_event(
+                    self.hass, self.entity_id, self.check_preset
+                )
+            )
+
+    @callback
+    async def check_preset(self, event: Event) -> None:
+        """Determine if the entity needs to be reloaded based on state change."""
+
+        if (
+            (new_state := event.data.get("new_state")) is None
+            or (old_state := event.data.get("old_state")) is None
+            or (new_preset := new_state.attributes.get("preset_mode")) is None
+            or (old_preset := old_state.attributes.get("preset_mode")) is None
+        ):
+            return
+
+        if old_preset != new_preset:
```

Comment:
Why do we need to reload when the preset state changes? We only need to reload if the set of supported presets change.

Reply:
The supported features change of every preset change, I could check supported features instead of presets but that's functionally equivalent.

Reply:
I'll rephrase, when the preset changes, the `_attr_supported_features` also changes. I originally made the change on the fly but was told that broke the rules.
See Frenck's comment https://github.com/home-assistant/core/pull/75246#issuecomment-1487116574
A reload is required every time the preset changes.

Reply:
Why does supported features change?

Reply:
Because `ClimateEntityFeature.TARGET_TEMPERATURE` and `ClimateEntityFeature.TARGET_TEMPERATURE_RANGE` are only supported on specific presets. For example, in the "MyTemp" preset, there is no temperature control on the main climate entity.

Reply:
Incase you need to know, this is why each preset has different features:
MyZone: you set the target temp for heat and cool
MyAuto: you set the target temp for heat and cool, and the temp range for heat_cool
MyTemp: you set the HVAC mode, and adjust the temperature using the other climate entities.

Reply:
We don't have a way at the moment to represent this in a good way for the user. The best we can do at the moment is to keep the all the supported features available and set both target temp and target temp high/low to `None` when selecting MyTemp preset and raise an error in the temperature service handlers during that time. The temperature control will be hidden in the more info view of the climate card. It will still be available in the main view of the climate card.

Reply:
Isnt this why @frenck said the integration should do a reload, that way only the supported features and HVAC modes of the currently running preset are avaliable? Anything else is a terrible UX or "breaks the rules".

I'd prefer to tell users to restart Home Assistant on preset change than give them every feature/HVAC mode of every preset all the time, especially since users rarely actually change presets.

Reply:
Supported features shouldn't change like this just by changing a mode of the entity. The climate entity currently lacks a way to well represent this preset and if we want to do that well we need to change the entity architecture design. We shouldn't try to work around the architecture.

We already have an HVAC mode where the user can't change temperature called `HVAC_AUTO`. This preset is similar but it's a preset instead of an HVAC mode and the user can change mode while in this preset. I'd start an architecture discussion with that preface and ask how can we represent this well and should we change the climate integration for this use case?